cms/djangoapps/contentstore/management/commands/tests/test_backfill_orgs_and_org_courses.py

```diff
@@ -0,0 +1,186 @@
+    @patch.object(organizations_api, 'bulk_add_organization_courses')
+    def test_arguments_and_input(
+            self,
+            mock_add_orgs,
+            mock_add_org_courses,
+            command_line_args,
+            user_inputs,
+            should_apply_changes,
+    ):
+        """
+        Test that the command-line arguments and user input processing works as
+        expected.
+
+        Given a list of `command_line_args` and a sequence of `user_inputs`
+        that will be supplied, we expect that:
+        * the user will be prompted a number of times equal to the length of `user_inputs`, and
+        * the command will/won't apply changes according to `should_apply_changes`.
+        """
+        with patch.object(
+            backfill_orgs_and_org_courses, "input", side_effect=user_inputs
+        ) as mock_input:
+            call_command("backfill_orgs_and_org_courses", *command_line_args)
+
+        # Make sure user was prompted the number of times we expected.
+        assert mock_input.call_count == len(user_inputs)
+
+        if should_apply_changes and user_inputs:
+            # If we DID apply changes and the user WAS prompted first,
+            # then we expect one DRY bulk-add run *and* one REAL bulk-add run.
+            assert mock_add_orgs.call_count == 2
+            assert mock_add_org_courses.call_count == 2
+            assert mock_add_orgs.call_args_list[0].kwargs == {"dry_run": True}
+            assert mock_add_org_courses.call_args_list[0].kwargs == {"dry_run": True}
+            assert mock_add_orgs.call_args_list[1].kwargs == {"dry_run": False}
+            assert mock_add_org_courses.call_args_list[1].kwargs == {"dry_run": False}
+        elif should_apply_changes:
+            # If DID apply changes but the user WASN'T prompted,
+            # then we expect just one REAL bulk-add run.
+            assert mock_add_orgs.call_count == 1
+            assert mock_add_org_courses.call_count == 1
+            assert mock_add_orgs.call_args.kwargs == {"dry_run": False}
+            assert mock_add_org_courses.call_args.kwargs == {"dry_run": False}
+        elif user_inputs:
+            # If we DIDN'T apply chnages but the user WAS prompted
+            # then we expect just one DRY bulk-add run.
+            assert mock_add_orgs.call_count == 1
+            assert mock_add_org_courses.call_count == 1
+            assert mock_add_orgs.call_args.kwargs == {"dry_run": True}
+            assert mock_add_org_courses.call_args.kwargs == {"dry_run": True}
+        else:
+            # Similarly, if we DIDN'T apply chnages and the user WASN'T prompted
+            # then we expect just one DRY bulk-add run.
+            assert mock_add_orgs.call_count == 1
+            assert mock_add_org_courses.call_count == 1
+            assert mock_add_orgs.call_args.kwargs == {"dry_run": True}
+            assert mock_add_org_courses.call_args.kwargs == {"dry_run": True}
```

Comment:
This code block could definitely be de-duped a bunch, but I prefer explicitly spelling out each code path that we're making assertions about.

Reply:
And yes, I see `chnages`.